apps/els_dap/src/els_dap_general_provider.erl

```diff
@@ -739,3 +729,17 @@ stop_debugger() ->
 -spec is_node_connected(node()) -> boolean().
 is_node_connected(Node) ->
   lists:member(Node, erlang:nodes(connected)).
+
+-spec safe_eval(node(), pid(), string(), update | no_update) -> term().
+safe_eval(ProjectNode, Debugged, Expression, Update) ->
+  {ok, Meta} = els_dap_rpc:get_meta(ProjectNode, Debugged),
+  Command = els_utils:to_list(Expression),
+  Return = els_dap_rpc:meta_eval(ProjectNode, Meta, Command),
+  case Update of
+    update -> ok;
+    no_update ->
```

Comment:
This is to prevent an endless update loop. Evaluating something in the context of the meta process triggers notifying the attached process.

For variable watching, we don't want to do this, because the client will trigger updating through evaluate request (which evaluate variables in the meta process and trigger an update, causing a loop).

This is the same for logging. Furthermore, we already resumed the process, and entering handle_info with int_cb is assumed to only happen on breakpoints.